src/main/java/seedu/address/model/event/StartDate.java

```
@@ -0,0 +1,143 @@
+    }
+
+    /**
+     * Returns true if a given string is a valid StartDate input.
+     * "" empty string is used to represent an empty StartDate.
+     * @return boolean
+     */
+
+    //found from https://mkyong.com/java/how-to-check-if-date-is-valid-in-java/
+    public static boolean isValidStartDate(String test) {
+        if (test == null) {
+            return true;
+        }
+        try {
+            LocalDate.parse(test, checkFormatter);
+        } catch (DateTimeParseException e) {
+            System.out.println("StartDate problem");
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Returns 1 if the other object is a StartDate that is later,
+     * -1 if the other object is a StartDate that is earlier,
+     * and 0 if the other object is a StartDate that is of the same date.
+     * @param other The object to compare with
+     * @return int
+     */
+    public int compareTo(Object other) {
+        if (other == null) {
+            return -1;
+        }
+        if (!(other instanceof StartDate)) {
+            throw new IllegalArgumentException(MESSAGE_ARGUMENT_CONSTRAINTS);
+        }
+        if (this.isEmpty() & ((StartDate) other).isEmpty()) {
+            return 0;
+        }
+        return this.date.compareTo(((StartDate) other).date);
+    }
+
+    /**
+     * Returns true if StartDate is empty, false otherwise
+     * @return boolean
+     */
+    public boolean isEmpty() {
+        return this.isEmpty;
+    }
+
+    /**
+     * Returns the String representation of the StartDate in format suitable for storage logging
+     * @return String
+     */
+    public String toLogFormat() {
+        if (this.isEmpty()) {
+            return null;
+        }
+        return this.date.format(logFormatter);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (!(other instanceof StartDate)) {
+            throw new IllegalArgumentException(MESSAGE_ARGUMENT_CONSTRAINTS);
```

`MESSAGE_ARGUMENT_CONSTRAINTS` is inappropriate here since this method is `equals()`, not `compareTo()`.

In fact, I don't think this is how `equals()` should be written, given how it is designed in other classes. Looks like it was copy-pasted from `compareTo()`. Also I don't believe `equals()` should throw any exceptions! 😆